node/network/bitfield-distribution/src/lib.rs

```diff
@@ -84,9 +86,8 @@ struct ProtocolState {
 	/// to determine what is relevant to them.
 	peer_views: HashMap<PeerId, View>,
 
-	/// Track all our neighbors in the current gossip topology.
-	/// We're not necessarily connected to all of them.
-	gossip_peers: HashSet<PeerId>,
+	/// The current gossip topology
+	topology: SessionGridTopology,
```

Comment:
We might consider using `SessionsGridTopologies` as in approval-distribution. 
This is more robust at session boundaries
cf https://github.com/paritytech/polkadot/issues/5316#issuecomment-1097754271

but requires some small changes to propagate messages received from a pending (future) topology upon topology change.

Reply:
It will also require to implement old topologies pruning (as currently this subsystem totally ignores finalized blocks messages).

Reply:
We can keep only current and previous topology, that should be enough. Not reference counting needed.

Reply:
What is the typical situation when there is a need in a previous topology: does it happen during a session change when we are propagating distribution for a previous session?

Reply:
Yes, there could active leaves (forks) at different heights and different sessions at session boundaries.

Reply:
Worst case if we don't handle multiple sessions is that the first few blocks of the sessions might not have as many b bitfields as needed, I think. Handling multiple sessions, if it's not too complex, is very much worth doing. but downsides aren't large because occupied availability cores are wiped on session changes. The stakes are much lower than approval-distribution.

Reply:
I have implemented a more simple approach suggested by @ordian: store just two topologies, one current and one for the previous session, if some other session is requested, then just a current topology is returned (as more or less sane default). 
This approach should cover the vast majority of the session boundaries issues I suppose.